### PR TITLE
Allow to run on local repos

### LINKED ISSUE #43: Run on local repositories
E.g.:
 - via cli string flag
 - via cli flag pointing to filename

Why:
 - will enable non-github workflows on top

### LINKED ISSUE #57: Apply fix to local repository instead of opening PR

### PR DESCRIPTION
**Pull request by klieret**
Independently specify a source of the problem statement (github issue URL or local file) and `repo_path` (github URL or local repo). If `repo_path` is not specified and the problem statement is from a github issue, then the corresponding github URL is used.

The new flag `--apply_patch_locally` can be used to apply the patch when working on a local repo.

Usage (updated)

```bash
# local issue, local repo
python run.py \
	...
	--repo_path=/path/to/swe-agent-test-repo/ \  
	--data_path=/path/to/swe-agent-test-repo/problem_statements/1.md  \
    --apply-patch_locally

# gh issue, local repo
python run.py \
	...
	--repo_path=/path/to/swe-agent-test-repo/ \  
	--data_path=https://github.com/klieret/swe-agent-test-repo/issues/1 \
	--apply_patch_locally

# local issue, gh repo
python run.py \
	...
	--repo_path=[/path/to/swe-agent-test-repo/](https://github.com/klieret/swe-agent-test-repo/) \  
	--data_path=/path/to/swe-agent-test-repo/problem_statements/1.md  \
	--open_pr

# gh issue, different gh repo
# also works. Will make another flag for the PRs unnecessary.
```

Closes #43 
Closes #57 

* [x] add more integration tests that run a whole replay
* [x] Make previous alt repo flag obsolete
* [x] Test with swe-bench style inputs

### CODE REVIEW

**Comment by ofirpress** [comment]
Exciting!

**Comment by klieret** [comment]
Upon thinking more, a lot of people probably will probably want to run over github issues, but locally. So I'm still debating of whether we instead should have `--data_path` point at the source of an issue (markdown file or issue URL) and instead have a `--repo_path` flag to specify a path to the repo.

**Comment by ofirpress** [comment]
I trust your instincts more than mine here, but I would also think that --repo_path is a better name than --data_path

File: run.py [hunk 1]
```diff
@@ -143,9 +148,11 @@ def main(args: ScriptArguments):
                 return_type="info_trajectory",
             )
             save_predictions(traj_dir, instance_id, info)
-            save_patch(traj_dir, instance_id, info)
+            patch_path = save_patch(traj_dir, instance_id, info)
             if args.actions.open_pr and should_open_pr(args, info, token=env._github_token):
                 env.open_pr(trajectory=trajectory, push_gh_repo_url=args.actions.push_gh_repo_url)
+            if args.actions.apply_patch_locally and patch_path is not None and env.record["repo"].startswith("local://"):
```
Comment (klieret) [inline run.py:154]: fixme

**Pull request merged**